crates/wasmparser/src/operators_validator.rs

```diff
@@ -561,17 +584,71 @@ impl OperatorValidator {
                 }
                 self.push_ctrl(FrameKind::Else, frame.block_type, resources)?;
             }
+            Operator::Try { ty } => {
+                self.check_exceptions_enabled()?;
+                self.check_block_type(ty, resources)?;
+                for ty in params(ty, resources)?.rev() {
+                    self.pop_operand(Some(ty))?;
+                }
+                self.push_ctrl(FrameKind::Try, ty, resources)?;
```

Comment:
More a note for later, but this, `Loop, `If`, and `Block` are all pretty similar so we may want to look into trying to unify those at some point. Not really need now, just a possibility!